src/buttons/UploadFileButton/UploadFileButton.tsx

```diff
@@ -0,0 +1,97 @@
+import { CloudUploadIcon } from 'lucide-react';
+
+import { CircularProgress, styled } from '@mui/material';
+
+import { ChangeEventHandler } from 'react';
+
+import GraaspButton, { GraaspButtonProps } from '../Button/Button';
+
+export interface UploadFileButtonProps {
+  id?: string;
+  size?: GraaspButtonProps['size'];
+  /**
+   * button color
+   */
+  color?: GraaspButtonProps['color'];
+  /**
+   * button variant
+   */
+  variant?: GraaspButtonProps['variant'];
+  /**
+   * callback when files are selected
+   */
+  onChange: ChangeEventHandler<HTMLInputElement>;
+  /**
+   * whether files are uploading
+   */
+  isLoading?: boolean;
+  /**
+   * text when loading
+   */
+  loadingText?: string;
+  /**
+   * text when changes have been detected
+   */
+  text?: string;
+
+  /**
+   * accept prop for browse button
+   */
+  accept?: string;
+  /**
+   * whether multiple files can be selected
+   */
+  multiple?: boolean;
+  /**
+   * button icon
+   */
+  icon?: JSX.Element | null;
+}
+
+const VisuallyHiddenInput = styled('input')({
+  clip: 'rect(0 0 0 0)',
+  clipPath: 'inset(50%)',
+  height: 1,
+  overflow: 'hidden',
+  position: 'absolute',
+  bottom: 0,
+  left: 0,
+  whiteSpace: 'nowrap',
+  width: 1,
+});
+
+const UploadFileButton = ({
+  id,
+  isLoading = false,
+  onChange,
+  loadingText = 'Uploading...',
+  text = 'Upload a file',
+  variant = 'contained',
+  accept,
+  multiple,
+  icon = <CloudUploadIcon />,
+  size,
+  color,
+}: UploadFileButtonProps): JSX.Element => (
+  <GraaspButton
+    id={id}
+    component='label'
+    variant={variant}
+    color={color}
+    startIcon={
```

Comment:
There is the `LoadingButton` that you can get from `@mui/lab` that does the loading indicator automatically.